src/anyio/_backends/_asyncio.py

```diff
@@ -2776,22 +2782,26 @@ async def wait_writable(cls, obj: HasFileno | int) -> None:
             obj = obj.fileno()
 
         if write_events.get(obj):
-            raise BusyResourceError("writing to") from None
+            raise BusyResourceError("writing to")
 
         loop = get_running_loop()
-        event = write_events[obj] = asyncio.Event()
-        loop.add_writer(obj, event.set)
+        event = asyncio.Event()
+        try:
+            loop.add_writer(obj, event.set)
+            remove_writer = loop.remove_writer
```

Comment:
Nit: I think this should be in the except's else clause

Reply:
I thought someone might pick up on that...

Reply:
Fixed.